### ISSUE TITLE
ManifoldSet API?

### ISSUE THREAD

**Issue by pca006132** [elalish/manifold#431]
From #425, I wonder if the best solution would be to provide a wrapper class for a group of manifolds. Every operation and transformation will be applied to each of the element inside the set, and at the end the user can get each of the parts out.
Everything can be done now, but this can probably make it a bit easier for users. And perhaps make a simple function that put each mesh in a way such that the bounding boxes are not overlapping, and can be imported into a 3D printer for example.

Example:
```js
const parts = ManifoldSet([
    cube([1, 2, 3]),
    cube([4, 5, 6])
])
parts = parts.subtract(cylinder(10, 1))
export(parts.sorted())
```

**Comment by elalish** [comment]
Yeah, I've been thinking about this too. I was thinking of an API exposed only within the editor for now, something to wrap the scene hierarchy of glTF. Something like:

```ts
class Node {
  parent?: Node;
  manifold?: Manifold;
  translation?: Vec3 | (t: number) => Vec3;
  rotation?: Vec3 | (t: number) => Vec3;
  scale?: Vec3 | (t: number) => Vec3;
  name?: string;
  constructor(): Node;
  clone(): Node;
}
```

Then if no `result` is defined we can just export the hierarchy of `Nodes` as a glTF scene, including rigid animations by using the functional versions. What do you think?

**Comment by pca006132** [comment]
This sounds good to me, but I am not sure about a couple of things:
1. For those transformations, why should we have a constant value version instead of just a constant function?
2. Should we have those different transformations separately, or just use a transformation matrix and then provide APIs to compose transformations together?
3. Will boolean operation of meshes with animation be supported?

**Comment by elalish** [comment]
1. Yes, that's an option - I was just thinking of API ergonomics. Let's discuss further.
2. They are animated separately (rotations can be interpolated with quaternion slerp, but interpolating a general 3x3 matrix is pretty undefined. The purpose of nodes in a scene graph is exactly to compose transformations.
3. No, the idea is operations go into creating the leaf nodes (where `node.manifold != null`). Animations are solely for rendering, so there's no need (or engine support) for manifold ops at that stage.

**Comment by elalish** [comment]
Also, regarding build-plate packing, that's a very complex and heuristic process (especially for powder printers). It's cool, but I feel like it's out of scope of this library. Would be great to demo connecting to a project that does it though. Actually, I'm pretty sure some print software (like Cura) has this built in anyway.

**Comment by pca006132** [comment]
Yeah, I was thinking about this because for stl you cannot specify multiple different parts, and decompose function in slicers require the parts to be disjoint which may not be the case here, and making an exploded view is probably even more complicated, so I was just thinking about a way to guarantee they are disjoint (not for the final printing)
But thinking about it, I think 3MF allows specifying multiple parts? So this should not be a concern.
